debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/xstream/XstreamStreamingChangeEventSource.java

```diff
@@ -71,37 +71,42 @@ public XstreamStreamingChangeEventSource(OracleConnectorConfig connectorConfig,
 
     @Override
     public void execute(ChangeEventSourceContext context) throws InterruptedException {
-        try {
-            // 1. connect
-            final byte[] startPosition = offsetContext.getLcrPosition() != null ? offsetContext.getLcrPosition().getRawPosition()
-                    : convertScnToPosition(offsetContext.getScn());
-            xsOut = XStreamOut.attach((oracle.jdbc.OracleConnection) jdbcConnection.connection(), xStreamServerName,
-                    startPosition, 1, 1, XStreamOut.DEFAULT_MODE);
-
-            LcrEventHandler handler = new LcrEventHandler(errorHandler, dispatcher, clock, schema, offsetContext, this.tablenameCaseInsensitive, this);
-
-            // 2. receive events while running
-            while (context.isRunning()) {
-                LOGGER.trace("Receiving LCR");
-                xsOut.receiveLCRCallback(handler, XStreamOut.DEFAULT_MODE);
+        try (OracleConnection xsConnection = new OracleConnection(jdbcConnection.config(), () -> getClass().getClassLoader())) {
```

Comment:
@Naros, are the changes in this class still needed after the removal of that call for obtaining the DB TZ?

Reply:
They are.  

For `TIMESTAMP WITH LOCAL TIME ZONE` data types, when we call `timestampValue` from the `TIMESTAMPLTZ` instance, the Oracle class executes a SQL call of `SELECT DBTIMEZONE FROM DUAL` which causes the concurrent use of the connection which leads to `oracle.streams.StreamsException: ORA-26824: user-defined XStream callback error`.

Reply:
Uh, so are you saying XStream itself is issue a DB SELECT whenever such timestamp value is retrieved?

Reply:
No this select is triggered when the converter detects the `TIMESTAMPLTZ` data type and we retrieve the actual timestamp value from the type so that the conversion can take place, see [here](https://github.com/debezium/debezium/pull/2225/files#diff-7130391480777348a00838e70123ef0a0344edcfe84a3bb124c00fd70bb85aafR521).

Reply:
Ok, but isn't this what we aimed to avoid by setting UTC as the connection/session TZ? So we know how to interpret such values?

Reply:
As discussed: let's just verify whether that call is done once and is cached then.

Reply:
The value is resolved once and cached.  I ran a snapshot with XStream with 10 rows, the first row triggered the query and then afterward no additional queries were figured to select the database's timezone.

Reply:
Ok, merging it then. Thx for verifying!